app/Http/Controllers/Secretariat/UserController.php

```diff
@@ -174,4 +176,67 @@ public function removeRole(Request $request, User $user, Role $role)
 
         return redirect()->back()->with('message', __('general.successful_modification'));
     }
+
+    /**
+     * Shows the page where a tenant can update their planned departure date.
+     */
+    public function showTenantUpdate()
+    {
+        if (!Auth::user()->needsUpdateTenantUntil()) {
+            return abort(403);
+        }
+        return view('user.update_tenant_status', [
+            'user' => Auth::user(),
+        ]);
+    }
+
+
+    /**
+     * Updates the planned departure date of a tenant.
+     */
+    public function updateTenantUntil(Request $request)
+    {
+        if (!Auth::user()->needsUpdateTenantUntil()) {
+            return abort(403);
+        }
+        $validator = Validator::make($request->all(), [
+            'tenant_until' => 'required|date|after:today',
+        ]);
+
+        if ($validator->fails()) {
+            return redirect()->back()
+                ->withErrors($validator)
+                ->withInput();
+        }
+
+        $user = Auth::user();
+        $date=Carbon::now()->addMonths(6);
+        $tenantUntilDate=Carbon::parse($request->tenant_until);
+        if (Carbon::now()->addMonths(6)->gt($tenantUntilDate)) {
+            $date = $tenantUntilDate;
+        }
+        $user->internetAccess()->update(['has_internet_until' => $date]);
+        $user->personalInformation()->update(['tenant_until' => $date]);
+
```

Comment:
Modify the Usercontroller::updatePersonalInformation() accordingly and use that instead of this.

Reply:
Can't it stay this way? I'll have to add another if in the redirect if I use another base url...

Other than that, if I modify the `updatePersonalInformation` function, we won't be able to mark everything required in the validator.

Reply:
If the validator can not make this, then my first attempt would be to include the updatePersonalinformation blade in this page. I worked hard to have only one user update function :D